third_party/move/move-compiler-v2/src/file_format_generator/module_generator.rs

```diff
@@ -568,6 +609,427 @@ impl ModuleGenerator {
         idx
     }
 
+    /// Common builder for struct apis
+    fn build_struct_api_index_common<K: Ord>(
+        &mut self,
+        ctx: &ModuleContext,
+        loc: &Loc,
+        struct_env: &StructEnv,
+        op_prefix: &str, // e.g., well_known::PACK / UNPACK / TEST_VARIANT
+        param_sig_index: Option<FF::SignatureIndex>,
+        return_sig_index: Option<FF::SignatureIndex>,
+        include_nonvariant_case: bool, // PACK/UNPACK: true; TEST_VARIANT: false
+        key_conv: fn(Option<Symbol>) -> K,
+    ) -> BTreeMap<K, FF::FunctionHandleIndex> {
+        let module = self.module_index(ctx, loc, &struct_env.module_env);
+        let struct_full_name = struct_env.get_full_name_with_address();
+        let fun_name_prefix = format!(
+            "{}{}{}",
+            op_prefix,
+            well_known::PUBLIC_STRUCT_DELIMITER,
+            struct_full_name
```

Comment:
Simple struct name should be sufficient, as the names are already scoped in the defining module.